docs/audio.rst

```diff
@@ -83,8 +86,215 @@ The built-in sounds can be called using ``audio.play(Sound.NAME)``.
+
+    .. py:attribute:: interpolation
+
+        The type of curve between the start and end
+        frequencies, different wave shapes have different rates of change
+        in frequency. One of the following values: ``INTER_LINEAR``,
+        ``INTER_CURVE``, ``INTER_LOG``.
+
+The arguments used to create any Sound Effect, including the built in effects,
+can be inspected by looking at each of the Effect instance attributes, or by
+converting the instance into a string (which can be done via ``str()``
+function, or by using a function that does the conversion automatically like
+``print()``).
+
+For example, with the :doc:`REPL </devguide/repl>` you can inspect the built
+in Effects::
+
+    >>> audio.Effect.CROAK
+    Effect(freq_start=..., freq_end=..., duration=..., vol_start=..., vol_end=..., wave=..., fx=..., interpolation=...)
+
+The built in Effects are immutable, so they cannot be changed. Trying to modify
+a built in Effect will throw an exception::
+
+    >>> audio.Effect.CLICK.duration = 1000
+    Traceback (most recent call last):
+        File "<stdin>", line 1, in <module>
+    TypeError: effect cannot be modified
+
+But a new one can be created like this::
+
+    >>> click_clone = Effect(audio.Effect.CLICK)
+    >>> click_clone.duration = 1000
+    >>>
+
+Built in Sound Effects
+----------------------
+
+Some pre-created Sound Effects are already available as examples. These can
+be played directly ``audio.play(audio.Effect.SQUEAK)``, or used as a base to
+create new effects ``audio.Effect(audio.Effect.SQUEAK, duration=2000)``.
+
+* ``audio.Effect.SQUEAK``
+* ``audio.Effect.WARBLE``
+* ``audio.Effect.CHIRP``
+* ``audio.Effect.CROAK``
+* ``audio.Effect.CLICK``
+
+Sound Effects Example
+---------------------
+
+::
+
+    from microbit import *
+
+    # Play a built in Sound Effect
+    audio.play(audio.Effect.CHIRP)
+
+    # Create a Sound Effect and immediately play it
+    audio.play(Effect(
+        freq_start=400,
+        freq_end=2000,
+        duration=500,
+        vol_start=100,
+        vol_end=255,
+        wave=audio.WAVE_TRIANGLE,
+        fx=audio.FX_VIBRATO,
+        interpolation=audio.LOG
+    ))
+
+    # Play a Sound Effect instance, modify an attribute, and play it again
+    my_effect = Effect(
+        preset=audio.CHIRP
+        freq_start=400,
+        freq_end=2000,
+    )
+    audio.play(my_effect)
+    my_effect.duration = 1000
+    audio.play(my_effect)
+
+    # You can also create a new effect based on an existing one, and modify
+    # any of its characteristics via arguments
+    audio.play(audio.Effect.WARBLE)
+    my_modified_effect = Effect(audio.Effect.WARBLE, duration=1000)
+    audio.play(my_modified_effect)
+
+    # Use sensor data to modify and play the existing Sound Effect instance
+    while True:
+        my_effect.freq_start=accelerometer.get_x()
+        my_effect.freq_end=accelerometer.get_y()
+        audio.play(my_effect)
```

Comment:
This is really nice and much clearer than also containing *all* the parameters and @microbit-giles I'd like us to make sure we include some nice lean examples in the edu content that encourage people to use this cleaner interface.